### PR TITLE
[BUBO-97] [BUBO-99] Judge review contests and judge rewards list

### PR DESCRIPTION
**Pull request by icedevera**
[[BUBO-97](https://vacuum.atlassian.net/browse/BUBO-97?atlOrigin=eyJpIjoiN2U4YmM1ZTIyYTI2NGE1OTk5ZTY5N2UwY2Q4Nzc3NzEiLCJwIjoiaiJ9)] [[BUBO-99](https://vacuum.atlassian.net/browse/BUBO-99?atlOrigin=eyJpIjoiN2Q0MGVlNjA0MzI5NDhhODk1MzFlY2ZiYTAxNjU3NTIiLCJwIjoiaiJ9)]

## Fixed and added flows for judges

**Fixed contest overview stats to properly show the pending tasks for judges:**
image

**Added Judge rewards page that shows contest distributed rewards and contest rewards to payout: (TODO: ability topayout rewards. Need to work on judging findings first)**
image

**Added button actions in contests table to make it clearer what the judge has to do for the contests:**
image
image

**Added the ability for judges to finalize/calculate rewards:**
image

**Altered judge review contest flow to properly use the statuses `ContestStatus.PENDING` and `ContestStatus.IN_REVIEW`. Added a new field `rewardTransferAddress` to the db to accomodate this flow:**

1. New contest is created (`ContestStatus.IN_REVIEW`). Judge can now review contest:
image

2. Judge adds `rewardTransferAddress` and marks contest as `ContestStatus.PENDING` (The reward transfer address is needed in order for the `proejct_owner` to know where to transfer the reward funds):
image

3. Project owner now has to transfer rewards and provide the `rewardTransferTxHash`: 
image
image

4. Judge checks `rewardTransferTxHash` and approves/rejects contest:
image

[BUBO-97]: https://vacuum.atlassian.net/browse/BUBO-97?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[BUBO-99]: https://vacuum.atlassian.net/browse/BUBO-99?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/sections/contest/ContestDetails/ContestInfo.tsx [hunk 1]
```diff
@@ -115,54 +136,78 @@ const ContestInfo = ({
...
+      <div className="mb-12 flex flex-col gap-3">
+        <h3 className="text-bodyL text-purple-light">
+          Rewards Transfer TX Hash
+        </h3>
+        {contest.rewardsTransferTxHash ? (
+          <Link
+            href={`https://cardanoscan.io/transaction/${contest.rewardsTransferTxHash}`}
+            className="text-bodyM underline">
+            {contest.rewardsTransferTxHash}
+          </Link>
```
Comment (matejfalat) [inline src/components/sections/contest/ContestDetails/ContestInfo.tsx:148]: Wouldn't it be better to wrap it in `<Button variant="link">` to use the same design for links as everywhere?

File: src/components/sections/contest/ContestDetails/ContestInfoJudge.tsx [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+  const {mutate: reviewContestMutate} = useReviewContest()
+
+  const form = useForm<FormValues>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      rewardsTransferAddress: '',
+    },
+  })
+
+  const [openMarkAsPending, setOpenMarkAsPending] = useState(false)
```
Comment (matejfalat) [inline src/components/sections/contest/ContestDetails/ContestInfoJudge.tsx:64]:
```suggestion
  const [isMarkAsPendingOpened, setIsMarkAsPendingOpened] = useState(false)
```

File: src/components/sections/judge/JudgeRewardsTable.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            className="gap-3">
+            Judge contests
+          </Link>
+        </Button>
+      </div>
+    )
+  }
+
+  return (
+    <div className="flex flex-grow flex-col bg-black px-24 pb-24 pt-8">
```
Comment (matejfalat) [inline src/components/sections/judge/JudgeRewardsTable.tsx:59]: I think this `JudgeRewardsTable` component should be just a `<Table>` and this outside wrapper with paddings would be better moved to the parent, but that's just a detail anyway 🤷

**Review by matejfalat — APPROVED** [review]
Didn't test it, but in general the code is looking good 👍

File: src/components/sections/contest/ContestDetails/ContestInfoJudge.tsx
[the remark this answers is not in the document]
Comment (icedevera) [reply src/components/sections/contest/ContestDetails/ContestInfoJudge.tsx:1]: I prefer `ContestInfoJudge` so it's next to `ContestInfo` in the file list 🤷

File: src/components/sections/judge/JudgeContestsTableRow.tsx [hunk 4]
```diff
@@ -20,14 +34,116 @@ const JudgeContestsTableRow = ({
...
+    }
+
+    if (
+      contestOccurence === ContestOccurence.PAST &&
+      contest.status === ContestStatus.FINISHED &&
+      contest.rewardsToPay > 0
+    ) {
+      return (
+        <Button asChild variant="outline" size="small">
+          <Link href="#" className="gap-2 text-buttonS">
```
[the remark this answers is not in the document]
Comment (icedevera) [reply src/components/sections/judge/JudgeContestsTableRow.tsx:126]: Yep, will add it in once the respective features get implemented

File: src/components/sections/contest/ContestDetails/ContestInfo.tsx (hunk 1 above)
Comment (icedevera) [reply src/components/sections/contest/ContestDetails/ContestInfo.tsx:148]: Sure 👍

**Pull request merged**